### PR TITLE
WIP: Production setup

### PR DESCRIPTION
**Pull request by cableman**
Please note that some of this stuff has to be re-done when going over to using github actions. But for now it's create to get the project rolling.

### CODE REVIEW

File: infrastructure/docker/php7.2-fpm/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+LABEL maintainer="ITK Dev <itkdev@mkb.aarhus.dk>"
+
+ENV PHP_OPCACHE_VALIDATE_TIMESTAMPS="0" \
+    PHP_OPCACHE_MAX_ACCELERATED_FILES="20000" \
+    PHP_OPCACHE_MEMORY_CONSUMPTION="192" \
+    PHP_OPCACHE_MAX_WASTED_PERCENTAGE="10" \
+    PHP_MAX_EXECUTION_TIME="30" \
+    PHP_MEMORY_LIMIT="128M" \
+    PHP_POST_MAX_SIZE="8M" \
+    PHP_UPLOAD_MAX_FILESIZE="2M"
```
Comment (turegjorup) [inline infrastructure/docker/php7.2-fpm/Dockerfile]: For the upload service this is a bit small but can be changed later.

**Review by turegjorup — APPROVED** [review]

File: infrastructure/docker/php7.2-fpm/Dockerfile (hunk 1 above)
Comment (cableman) [reply infrastructure/docker/php7.2-fpm/Dockerfile]: Yes, this is the generic image and not done yet. This is made as env variables, so they can be overriden for the app images :-)

File: infrastructure/k8s/readme.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
+# Azure kubernetes service
+This document explain how-to configure AKS to run cover service in kubernetes with virtual node scaling and monitoring.
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: "This document explains how to configure AKS to run Cover Service in Kubernetes with virtual node scaling and monitoring."

File: infrastructure/k8s/readme.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
+# Azure kubernetes service
+This document explain how-to configure AKS to run cover service in kubernetes with virtual node scaling and monitoring. 
+
+Login in through CLI and list available regions. You should use a region in EU to ensure data safety.
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: "the EU"

File: infrastructure/k8s/readme.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+Set basic configuration variables used in this guide more than once. They are only set in the current terminal window used and only as long as it is not closed.
+```sh
+ksname=ddb-cover-service
+res=CoverService 
+region=westeurope
+version=$(az aks get-versions -l ${region} --query 'orchestrators[-1].orchestratorVersion' -o tsv)
+```
+
+## Create services and the cluster
+This setup is the more advanced AKS setup with virtual node (https://docs.microsoft.com/en-us/azure/aks/virtual-nodes-cli) to better handle peak traffic. This requires an AKS setup with advanced network setup to enabled communication between ACI (Azure container instance) and the cluster.
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: "to handle peak traffic better."

File: infrastructure/k8s/readme.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+```sh
+az provider register --namespace Microsoft.ContainerInstance
+```
+
+Create resources group 
+```sh
+az group create --name ${res} --location ${region}
+```
+
+Create a VNET with two subnets that going to be used
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: "Create a VNET with the two subnets that are going to be used"

File: infrastructure/k8s/readme.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+```sh
+az network vnet create \
+    --resource-group ${res} \
+    --name coverServiceVnet \
+    --address-prefixes 10.0.0.0/8 \
+    --subnet-name coverServiceSubnet \
+    --subnet-prefix 10.240.0.0/16
+```
+
+And an additional subnet
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: Why is this added?

File: infrastructure/k8s/readme.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+```
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: \```sh

File: infrastructure/k8s/readme.md [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+az network vnet subnet create \
+    --resource-group ${res} \
+    --vnet-name coverServiceVnet \
+    --name coverServiceVirtualNodeSubnet \
+    --address-prefix 10.241.0.0/16
+```
+
+To connect to ACI you need a service principal assignment. So create the principal first.
+```
+az ad sp create-for-rbac --skip-assignment
```
Comment (tuj) [inline infrastructure/k8s/readme.md:55]: \```sh

File: infrastructure/k8s/readme.md [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+```
+
+From the output from the above command we need to save the application id and password (app secret).
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: "From the output of the previous command

File: infrastructure/k8s/readme.md [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+```sh
+az role assignment create --assignee $APPID --scope $VNETID --role Contributor
+```
+
+Get the Azure subnet id.
+```sh
+SUBNET=$(az network vnet subnet show --resource-group ${res} --vnet-name coverServiceVnet --name coverServiceSubnet --query id -o tsv)
+```
+
+Create the cluster and wait for it to create the 3 nodes we have as your standard cluster (it takes some time, so get you self some coffe).
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: "Create the cluster and wait for it to create the 3 nodes in the standard cluster (it takes a bit of time, so go ahead and get yourself a cup of coffee)."

File: infrastructure/k8s/readme.md [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+    --addons virtual-node \
+    --subnet-name coverServiceVirtualNodeSubnet
+````
+
+Configure kubeclt to connect to the new cluster
+```sh
+az aks get-credentials --resource-group ${res} --name ${ksname}
+```
+
+Verify that you now are connected to the cluster.
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: "Verify that you are connected to the cluster now."

File: infrastructure/k8s/readme.md [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+```sh
+kubectl get nodes
+```
+
+## Helm
+We are going to use https://helm.sh/ to install ingress and cert-manager into the cluster setup. Note that we here are using helm version 3. We also install the kubectx helper tool as it makes switching cluster and namespaces easier.
+```
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: \```sh 
Apply to rest of readme

File: infrastructure/k8s/readme.md [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+
+Next enable the monitoring add-on by running the command below.
+```
+az aks enable-addons --resource-group ${res} \
+    --name ${ksname} \
+    --addons monitoring \
+    --workspace-resource-id ${workspaceresoid}
+```
+
+
```
Comment (tuj) [inline infrastructure/k8s/readme.md:203]: Empty line

File: infrastructure/k8s/readme.md [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+  apiGroup: rbac.authorization.k8s.io
+```
+
+```
+kubectl apply -f logreader-rbac.yaml
+```
+
+## Horizontal pod autoscaler (HPA)
+
+@TODO: description missing
```
Comment (tuj) [inline infrastructure/k8s/readme.md]: Missing

**Review by tuj — CHANGES_REQUESTED** [review]

File: infrastructure/k8s/readme.md (hunk 2 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Fixed

File: infrastructure/k8s/readme.md (hunk 3 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Fixed

File: infrastructure/k8s/readme.md (hunk 4 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Fixed

File: infrastructure/k8s/readme.md (hunk 5 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Fixed

File: infrastructure/k8s/readme.md (hunk 6 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: shhh... because the MS network AKS virtual node guide said that you have to.... so....

Comment (cableman) [reply infrastructure/k8s/readme.md]: :-)

File: infrastructure/k8s/readme.md (hunk 7 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: fixed

File: infrastructure/k8s/readme.md (hunk 8 above)
Comment (cableman) [reply infrastructure/k8s/readme.md:55]: fixed

File: infrastructure/k8s/readme.md (hunk 9 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Fixed

File: infrastructure/k8s/readme.md (hunk 10 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Fixed

File: infrastructure/k8s/readme.md (hunk 11 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Fixed

File: infrastructure/k8s/readme.md (hunk 12 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Fixed

File: infrastructure/k8s/readme.md (hunk 13 above)
Comment (cableman) [reply infrastructure/k8s/readme.md:203]: Fixed

File: infrastructure/k8s/readme.md (hunk 14 above)
Comment (cableman) [reply infrastructure/k8s/readme.md]: Yeah... not done that part in to AKS yet... so more writing will come here.

**Review by tuj — APPROVED** [review]
